sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -153,15 +163,47 @@ private static BiFunction<HttpClientRequest, NettyOutbound, Publisher<Void>> bod
                     hdr.getValuesList().forEach(value -> reactorNettyRequest.addHeader(hdr.getName(), value));
                 }
             }
-            if (restRequest.getBody() != null) {
-                Flux<ByteBuf> nettyByteBufFlux = restRequest.getBody().map(Unpooled::wrappedBuffer);
-                return reactorNettyOutbound.send(nettyByteBufFlux);
+            BinaryData body = restRequest.getBodyAsBinaryData();
+            if (body != null) {
+                BinaryDataContent bodyContent = BinaryDataHelper.getContent(body);
+                if (bodyContent instanceof ByteArrayContent) {
+                    return reactorNettyOutbound.send(Mono.just(Unpooled.wrappedBuffer(bodyContent.toBytes())));
+                } else if (bodyContent instanceof StringContent
+                    || bodyContent instanceof SerializableContent) {
+                    return reactorNettyOutbound.send(
+                        Mono.fromSupplier(() -> Unpooled.wrappedBuffer(bodyContent.toBytes())));
+                } else if (bodyContent instanceof FileContent) {
+                    FileContent fileContent = (FileContent) bodyContent;
+                    // Beware of NettyOutbound.sendFile(Path) it involves extra file length lookup.
+                    // This is going to use zero-copy transfer if there's no ssl
+                    return reactorNettyOutbound.sendFile(fileContent.getFile(), 0, fileContent.getLength());
+                } else if (bodyContent instanceof InputStreamContent) {
+                    return sendInputStream(reactorNettyOutbound, (InputStreamContent) bodyContent);
+                } else {
+                    Flux<ByteBuf> nettyByteBufFlux = restRequest.getBody().map(Unpooled::wrappedBuffer);
+                    return reactorNettyOutbound.send(nettyByteBufFlux);
+                }
             } else {
                 return reactorNettyOutbound;
             }
         };
     }
 
+    private static NettyOutbound sendInputStream(NettyOutbound reactorNettyOutbound, InputStreamContent bodyContent) {
+        return reactorNettyOutbound.sendUsing(
+            bodyContent::toStream,
```

Comment:
What happens in the case of retry and the `InputStream` is not replayable?

Reply:
It's going to be caller responsibility to rewind body before next attempt. Reactor-Netty doesn't attempt to retransmit if headers or body were already in flight.